### PR TITLE
feature/get-data-from-ttn

### LINKED ISSUE #219: Getting data from TTN

### PR DESCRIPTION
**Pull request by Mohammad96Assaf**
close #219

### CODE REVIEW

File: internal/storage/postgres/seed/99_demo_data.sql [hunk 1]
```diff
@@ -35,6 +35,9 @@ INSERT INTO sensors (id, status, latitude, longitude, geometry)
 VALUES
     ('sensor-4', 'online', 54.82078826498143, 9.489684366114483, ST_SetSRID(ST_MakePoint(54.82078826498143, 9.489684366114483), 4326));
 
+INSERT INTO sensors (id, status, latitude, longitude, geometry)
+VALUES
+    ('tree-sensor', 'online', 54.774932, 9.450000, ST_SetSRID(ST_MakePoint(54.774932, 9.450000), 4326));
```
Comment (choffmann) [inline internal/storage/postgres/seed/99_demo_data.sql:40]: I don't know exactly what the purpose of this is, but ok

**Review by choffmann — CHANGES_REQUESTED** [review]

File: internal/service/domain/sensor/mqtt.go [hunk 2]
```diff
@@ -3,23 +3,81 @@ package sensor
...
 
+	"github.com/go-playground/validator/v10"
 	domain "github.com/green-ecolution/green-ecolution-backend/internal/entities"
+	"github.com/green-ecolution/green-ecolution-backend/internal/service"
 	"github.com/green-ecolution/green-ecolution-backend/internal/storage"
+	storageSensor "github.com/green-ecolution/green-ecolution-backend/internal/storage/postgres/sensor"
+	"github.com/pkg/errors"
 )
 
 type MqttService struct {
```
[the remark this answers is not in the document]
Comment (Mohammad96Assaf) [reply internal/service/domain/sensor/mqtt.go:14]: It is a good idea, but let's do it later in a new feature branch. New changes will be made to these files, which are included in other merge requests, such as mapping the sensor to the tree and updating the sensor status.

File: internal/storage/postgres/seed/99_demo_data.sql (hunk 1 above)
Comment (Mohammad96Assaf) [reply internal/storage/postgres/seed/99_demo_data.sql:40]: It was a test sensor to check if the ID "tree-sensor" from TTN exists or not.

File: internal/service/domain/sensor/mqtt.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (doriengr) [reply internal/service/domain/sensor/mqtt.go:14]: @Mohammad96Assaf Let's create a new issue for it 👍 Can you create one, please 🙏?

File: internal/storage/postgres/sensor/create.go [hunk 3]
```diff
@@ -82,12 +82,22 @@ func (r *SensorRepository) InsertSensorData(ctx context.Context, data []*entitie
...
-		Status:    sqlc.SensorStatus(sensor.Status),
+	id, err := r.store.CreateSensor(ctx, &sqlc.CreateSensorParams{
+		ID:     sensor.ID,
+		Status: sqlc.SensorStatus(sensor.Status),
+	})
+	if err != nil {
+		return "", err
+	}
+
+	if err := r.store.SetSensorLocation(ctx, &sqlc.SetSensorLocationParams{
```
Comment (doriengr) [inline internal/storage/postgres/sensor/create.go:93]: I am not quite sure why the sensor location is set after the sensor is created? Is

File: internal/service/domain/sensor/mqtt.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mohammad96Assaf) [reply internal/service/domain/sensor/mqtt.go:14]: done! 👍

File: internal/storage/postgres/sensor/create.go (hunk 3 above)
Comment (Mohammad96Assaf) [reply internal/storage/postgres/sensor/create.go:93]: Because of the geometry attribute.

**Review by doriengr — APPROVED** [review]

File: internal/service/domain/sensor/mqtt.go [hunk 4]
```diff
@@ -3,23 +3,81 @@ package sensor
...
-func (s *MqttService) HandleMessage(_ context.Context, _ *domain.MqttPayload) (*domain.MqttPayload, error) {
-	// TODO: Implement the business logic of HandleMessage
+func (s *MqttService) HandleMessage(ctx context.Context, payload *domain.MqttPayload) ([]*domain.SensorData, error) {
+	if payload == nil {
+		return nil, handleError(errors.New("mqtt payload is nil"))
+	}
 
-	return nil, nil
+	if err := s.validator.Struct(payload); err != nil {
+		return nil, service.NewError(service.BadRequest, errors.Wrap(err, service.ErrValidation.Error()).Error())
```
[the remark this answers is not in the document]
Comment (choffmann) [reply internal/service/domain/sensor/mqtt.go]: This note has not been implemented, please do not use the service error here

**Review by choffmann — APPROVED** [review]

**Pull request merged**